src/librustc_data_structures/binary_search_util/mod.rs

```diff
@@ -15,34 +15,49 @@ where
         Err(_) => return &[],
     };
 
-    // We get back *some* element with the given key -- so
-    // search backwards to find the *first* one.
-    //
-    // (It'd be more efficient to use a "galloping" search
-    // here, but it's not really worth it for small-ish
-    // amounts of data.)
-    let mut start = mid;
-    while start > 0 {
-        if key_fn(&data[start - 1]) == *key {
-            start -= 1;
-        } else {
+    // We get back *some* element with the given key -- so do
+    // a galloping search backwards to find the *first* one.
+    let (mut start, mut step, mut previous, size) = (mid, 1, mid, data.len());
```

Comment:
Could we declare these on separate lines? My eyes are having trouble correlating initial values with the variable names :)